qiskit_aer/quantum_info/states/aer_statevector.py

```diff
@@ -179,34 +193,65 @@ def _aer_evolve_circuit(aer_state, circuit, qubits):
 
     @staticmethod
-    def _aer_evolve_instruction(aer_state, inst, qubits):
+    def _aer_evolve_instruction(aer_state, inst, qubits, basis_gates=None):
         """Apply instruction into aer_state"""
+
         params = inst.params
-        if inst.name in ['u', 'u3']:
-            aer_state.apply_mcu(qubits[0:len(qubits) - 1], qubits[len(qubits) - 1],
-                                params[0], params[1], params[2])
-        elif inst.name in ['x', 'cx', 'ccx']:
-            aer_state.apply_mcx(qubits[0:len(qubits) - 1], qubits[len(qubits) - 1])
-        elif inst.name in ['y', 'cy']:
-            aer_state.apply_mcy(qubits[0:len(qubits) - 1], qubits[len(qubits) - 1])
-        elif inst.name in ['z', 'cz']:
-            aer_state.apply_mcz(qubits[0:len(qubits) - 1], qubits[len(qubits) - 1])
-        elif inst.name == 'unitary':
-            aer_state.apply_unitary(qubits, inst.params[0])
-        elif inst.name == 'diagonal':
-            aer_state.apply_diagonal(qubits, inst.params[0])
+        applied = True
+
+        if basis_gates and inst.name in basis_gates:
+            if inst.name in ['u3', 'u']:
+                aer_state.apply_u(qubits[0], params[0], params[1], params[2])
+            elif inst.name == 'h':
+                aer_state.apply_h(qubits[0])
+            elif inst.name == 'x':
+                aer_state.apply_x(qubits[0])
+            elif inst.name == 'cx':
+                aer_state.apply_cx(qubits[0], qubits[1])
+            elif inst.name == 'y':
+                aer_state.apply_y(qubits[0])
+            elif inst.name == 'cy':
+                aer_state.apply_cy(qubits[0], qubits[1])
+            elif inst.name == 'z':
+                aer_state.apply_z(qubits[0])
+            elif inst.name == 'cz':
+                aer_state.apply_cz(qubits[0], qubits[1])
+            elif inst.name == 'unitary':
+                aer_state.apply_unitary(qubits, inst.params[0])
+            elif inst.name == 'diagonal':
+                aer_state.apply_diagonal(qubits, inst.params)
+            elif inst.name == 'cu':
+                aer_state.apply_cu(qubits[0], qubits[1], params[0], params[1], params[2], params[3])
+            elif inst.name == 'mcu':
+                aer_state.apply_mcu(qubits[0:len(qubits) - 1], qubits[len(qubits) - 1],
+                                    params[0], params[1], params[2], params[3])
+            elif inst.name in 'mcx':
+                aer_state.apply_mcx(qubits[0:len(qubits) - 1], qubits[len(qubits) - 1])
+            elif inst.name in 'mcy':
+                aer_state.apply_mcy(qubits[0:len(qubits) - 1], qubits[len(qubits) - 1])
+            elif inst.name in 'mcz':
+                aer_state.apply_mcz(qubits[0:len(qubits) - 1], qubits[len(qubits) - 1])
+            elif inst.name == 'id':
+                pass
+            else:
+                applied = False
+        elif inst.name == 'kraus':
+            aer_state.apply_kraus(qubits, inst.params)
```

Comment:
It seems a bit unusual to allow a statevector to apply a Kraus operation, outside the context of a full simulator with managed repeats, etc.  I guess it's probably ok because we already support `reset` (which Terra's `Statevector` also does), which is also probabilistic?

Feels weird to me, but if it's useful, I'm not particularly against it.

Reply:
@jakelishman thank you for your comments. Yes, `q.qi.Statevector` supports `reset`, which does not apply unitary. kraus operator is useful to implement simulator by using quantum_info with noise model.